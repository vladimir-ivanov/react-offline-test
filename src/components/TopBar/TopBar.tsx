import React, { FunctionComponent } from 'react';
import { createStyles, makeStyles } from '@material-ui/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { CssBaseline } from '@material-ui/core';

interface Props {
    title: string;
}

const useStyles = makeStyles(() =>
    createStyles({
        root: {
            flexGrow: 1
        }
    })
);

export const TopBar: FunctionComponent<Props> = ({ title }) => {
    const classes = useStyles({});

    return (
        <>
            <CssBaseline />
            <div className={classes.root}>
                <AppBar position="static">
                    <Toolbar>
                        <Typography variant="h6">
                            {title}
                        </Typography>
                    </Toolbar>
                </AppBar>
            </div>
        </>
    );
};
