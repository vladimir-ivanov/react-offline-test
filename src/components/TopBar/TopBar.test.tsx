import '../../setupEnzyme';

import * as React from 'react';
import { mount } from 'enzyme';
import { TopBar } from './TopBar';

jest.mock('@material-ui/core',() => {
    return {CssBaseline: () => (<>CssBaseline</>)};
});

//TODO - snapshot serialization needs fixing, does not work correctly
describe('TopBar', () => {
    it('should render correctly', () => {
        expect(mount(<TopBar title='some title' />)).toMatchSnapshot();
    });
});
