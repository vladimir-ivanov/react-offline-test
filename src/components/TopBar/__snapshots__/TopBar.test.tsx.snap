// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar should render correctly 1`] = `ReactWrapper {}`;
